src/app/services/log.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { LogLevel } from '../models/log-level.model';
+
+@Injectable({
```

Comment:
I'm far from an Angular expert, but I don't think this needs to be `Injectable` does it? Couldn't this just be a regular "utility" class?

Reply:
I'm not entirely sure either, I'm curious how that would affect testing since` log.service.spec.ts` currently uses the Angular [TestBed](https://angular.io/api/core/testing/TestBed). But I agree that it's not really necessary for it to be an Injectable, and I could potentially refactor the LogService test suite to remove the references to `TestBed`.

Reply:
So I don't think there is any reason to use the Angular TestBed in this case. Just import the class and test it "normally". If you need any help with it, just let me know, I'd be happy to help.

Reply:
No such thing as a "silly question"! What do you mean by a "service" class? In my mind, a "service" class in this case is just an aggregation of similar functions as well as an abstraction layer. In other words, a "service" class is just a name. Does that make sense? Happy to discuss further

Reply:
That definitely makes sense, and I appreciate the answer! I think I was worried about the file still having the `.service.ts` file extension now that it's not longer considered an [Angular Injectable service class](https://angular.io/guide/architecture-services#introduction-to-services-and-dependency-injection). But even the service examples in the documentation I linked have an almost identical `Logger` service that doesn't use a `@Injectable()` 😅